src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -361,27 +361,52 @@ case class JesBackend(actorSystem: ActorSystem)
-    val log = workflowLoggerWithCall(backendCall)
-    authenticateAsUser(backendCall.workflowDescriptor) { interface =>
-      Try(interface.copy(cachedCall.callGcsPath.toString, backendCall.callGcsPath.toString)) match {
-        case Failure(ex) =>
-          log.error(s"Exception occurred while attempting to copy outputs from ${cachedCall.callGcsPath} to ${backendCall.callGcsPath}", ex)
-          FailedExecutionHandle(ex).future
-        case Success(_) => postProcess(backendCall) match {
-          case Success(outputs) => backendCall.hash map { h =>
-            SuccessfulExecutionHandle(outputs, Seq.empty[ExecutionEventEntry], backendCall.downloadRcFile.get.stripLineEnd.toInt, h, Option(cachedCall)) }
-          case Failure(ex: AggregatedException) if ex.exceptions collectFirst { case s: SocketTimeoutException => s } isDefined =>
-            // TODO: What can we return here to retry this operation?
-            // TODO: This match clause is similar to handleSuccess(), though it's subtly different for this specific case
-            val error = "Socket timeout occurred in evaluating one or more of the output expressions"
-            log.error(error, ex)
-            FailedExecutionHandle(new Throwable(error, ex)).future
-          case Failure(ex) => FailedExecutionHandle(ex).future
-        }
+  def useCachedCall(cachedCall: BackendCall, backendCall: BackendCall)(implicit ec: ExecutionContext): Future[ExecutionHandle] = {
+    import better.files._
+
+    def renameCallSpecificFiles = {
+      val namesBuilder = List(
+        JesBackendCall.jesLogStdoutFilename _,
+        JesBackendCall.jesLogStderrFilename _,
+        JesBackendCall.jesLogFilename _,
+        JesBackendCall.jesReturnCodeFilename _
+      )
+
+      val wrongPaths = namesBuilder map { _(cachedCall.key) } map backendCall.callGcsPath.resolve
+      val correctNames = namesBuilder map { _(backendCall.key) }
+
+      (wrongPaths zip correctNames) map {
+        case (wrongPath, correctName) => wrongPath.renameTo(correctName)
       }
     }
-  } flatten
+
+    def copyingWork(gcs: GoogleCloudStorage) = for {
+        _ <- Try(gcs.copy(cachedCall.callGcsPath.toString, backendCall.callGcsPath.toString))
+        _ <- Try(renameCallSpecificFiles)
+    } yield ()
+
+    Future {
+      val log = workflowLoggerWithCall(backendCall)
+      authenticateAsUser(backendCall.workflowDescriptor) { interface =>
+        copyingWork(interface) match {
+          case Failure(ex) =>
+            log.error(s"Exception occurred while attempting to copy outputs from ${cachedCall.callGcsPath} to ${backendCall.callGcsPath}", ex)
+            FailedExecutionHandle(ex).future
+          case Success(_) => postProcess(backendCall) match {
+            case Success(outputs) => backendCall.hash map { h =>
+              SuccessfulExecutionHandle(outputs, Seq.empty[ExecutionEventEntry], backendCall.downloadRcFile.get.stripLineEnd.toInt, h, Option(cachedCall)) }
```

Comment:
I realize this is copypasta but can you make that List.empty or replace w/ a more appropriate collection type if list isn't right?